cmd/kubeadm/app/componentconfigs/checksums.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package componentconfigs
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"sort"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+// ChecksumAnnotation holds the object annotation key that kubeadm uses to store a SHA256 sum to verify for user changes
+const ChecksumAnnotation = "kubeadm.kubernetes.io/autogen-checksum"
```

Comment:
@fabriziopandini recommended here
https://github.com/kubernetes/enhancements/pull/1725/files#r426238092
using the name `hash`. are we still going for that?

also `*.kubernetes.io` is technically reserved for core:
https://github.com/kubernetes/community/blob/master/sig-architecture/api-review-process.md#voluntary

but kubeadm sort of reserved its usage with the API being `kubeadm.k8s.io` and existing before `*.x-k8s.io/` was established

we also have these already:
https://github.com/kubernetes/kubernetes/blob/89ba90573f163ee3452b526f30348a035d54e870/cmd/kubeadm/app/constants/constants.go#L380
https://github.com/kubernetes/kubernetes/blob/89ba90573f163ee3452b526f30348a035d54e870/cmd/kubeadm/app/constants/constants.go#L383
https://github.com/kubernetes/kubernetes/blob/89ba90573f163ee3452b526f30348a035d54e870/cmd/kubeadm/app/constants/constants.go#L215

so i suggest we stick to `kubeadm.kubernetes.io`, unless we get complains from sig-arch (which is not that unlikely).

Reply:
I really don't care that much what this is going to be called. However, we need to settle on something, because I already have several different suggestions here (probably a couple of which made by @fabriziopandini ).
In fact, I think that other than this name here, everything else should be ready and already agreed upon in the past.
I might as well raise the issue tomorrow in the office hours if we haven't settled on a name yet.

Reply:
ok, let's discuss tomorrow.

for hash vs checksum, this search reveals "hash" is the more used term:
https://github.com/kubernetes/kubernetes/search?q=path%3A%2Fcmd%2Fkubeadm+hash&unscoped_q=path%3A%2Fcmd%2Fkubeadm+hash

https://github.com/kubernetes/kubernetes/search?q=path%3A%2Fcmd%2Fkubeadm+checksum&unscoped_q=path%3A%2Fcmd%2Fkubeadm+checksum